### PR TITLE
140 tabchange evnt

### LINKED ISSUE #140: Implement TabChangeEvent
This should be done in the MainController. In addition to the semantic event Start/stop-watching-PR, this tab-change event can be logged on any tab change.

### PR DESCRIPTION
**Pull request by MathiasMeuleman**
Will close #140 

Added a TabchangeEvent. Is fired whenever a tab is changed (for obvious reasons)

### CODE REVIEW

**Comment by Exclaminator** [comment]
LGTM

**Comment by mpsijm** [comment]
When I use Ctrl+Tab and flip through my entire tab list (you know how many I have :innocent: ) it doesn't log much? :P

**Comment by MathiasMeuleman** [comment]
Of course it doesn't, the plugin doesn't log anything when the user is on other tabs. The only thing that can and should be logged is a tab change from or to the PR.

**Comment by mpsijm** [comment]
I'm not sure whether that was the initial idea in that case :P I thought we needed to log all tab changes. Maybe check the meeting minutes?

**Comment by MathiasMeuleman** [comment]
Meeting minutes are inconslusive. But, since only a timestamp is logged and not the previous tab/next tab, it is absolutely not useful to log every tab change, since this doesn't provide any information.

**Comment by mpsijm** [comment]
It indeed doesn't matter whether next/previous tab is logged, I only meant to say that I think that all tab changes should be logged, and Ctrl+Tabbing through my tabs was an easy way to figure out that that currently does not happen :

File: src/octopeer-github/main/Database/EventObject/TabChangeEvent.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Created by Mathias on 2016-06-15.
+ */
+
+/**
```
Comment (thervh70) [inline src/octopeer-github/main/Database/EventObject/TabChangeEvent.ts]: Split comment. Please merge :P

**Comment by thervh70** [comment]
Please fix the comment issue stated in the code. Then it is ready to merge!

File: src/octopeer-github/main/Database/EventObject/TabChangeEvent.ts (hunk 1 above)
Comment (MathiasMeuleman) [inline src/octopeer-github/main/Database/EventObject/TabChangeEvent.ts]: This is consistent over all EventObjects

Comment (mpsijm) [inline src/octopeer-github/main/Database/EventObject/TabChangeEvent.ts]: But not consistent over the entire project :wink:

**Comment by mdingjan** [comment]
I agree with the comments that already have been placed. Make sure to fix these, then this PR can be merged.

File: src/octopeer-github/main/Database/EventObject/TabChangeEvent.ts (hunk 1 above)
Comment (MathiasMeuleman) [inline src/octopeer-github/main/Database/EventObject/TabChangeEvent.ts]: You are one to speak @mpsijm, you are the one who made it consistent here

Comment (MathiasMeuleman) [inline src/octopeer-github/main/Database/EventObject/TabChangeEvent.ts]: Done

**Comment by mdingjan** [comment]
Thanks for processing the feedback; will merge!

**Pull request merged**